google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudApiManagementConfirmationDialog.java

```diff
@@ -174,9 +174,23 @@ Path getServiceAccountKeyDownloadPath() {
   @Override
   protected ValidationInfo doValidate() {
     if (createNewServiceAccountCheckbox.isSelected()) {
-      if (StringUtils.isEmpty(serviceAccountNameTextField.getText())) {
+      String name = serviceAccountNameTextField.getText().trim();
+
+      if (StringUtils.isEmpty(name)) {
+        return new ValidationInfo(
+            GctBundle.message("cloud.apis.management.dialog.serviceaccount.name.empty.error"),
+            serviceAccountNameTextField);
+      } else if (name.length() > CloudApiManager.SERVICE_ACCOUNT_NAME_MAX_LEN) {
+        return new ValidationInfo(
+            GctBundle.message(
+                "cloud.apis.management.dialog.serviceaccount.name.len.error",
+                CloudApiManager.SERVICE_ACCOUNT_NAME_MAX_LEN),
+            serviceAccountNameTextField);
+      } else if (!CloudApiManager.SERVICE_ACCOUNT_ID_PATTERN.matcher(name).matches()) {
         return new ValidationInfo(
-            GctBundle.message("cloud.apis.management.dialog.serviceaccount.name.error"),
+            GctBundle.message(
+                "cloud.apis.management.dialog.serviceaccount.name.regex.error",
+                CloudApiManager.SERVICE_ACCOUNT_ID_PATTERN.toString()),
```

Comment:
It could be a bit unfriendly to show a regular expression when error is saying service account name is not valid, could we show something similar in wording, i.e. 'contains only characters, numbers etc.'?

Reply:
I thought about this, but there are other rules, like it cannot start with a number etc. This could get unwieldy to put in words so I just used the regex (since these are developers anyway). wdyt?

Reply:
Might be a good question for UX person... I can image a scenario when something is wrong with the name, someone is shown a regex which he doesn't understand and it's kind of guessing game.

Reply:
hmm. yeah I definitely see your concern. I'm tempted to leave it. @patflynn any opinion on this?

Reply:
I kind of like a regular expressions expressivity, but I agree with @ivanporty that we can't expect our users to understand a regular expression.  I'd be OK with a verbal description and/or link to docs on what is valid.

Reply:
ok. I'll use wording instead. I don't think this exists in the docs (i'll look). I discovered this pattern by executing (failed) API calls and seeing the message in the API response which printed out this regex.